### PR TITLE
return default transform for axes in bivariate_matrix api

### PR DESCRIPTION
**Pull request by sleeping-h**
## Summary by CodeRabbit

- **New Features**
  - Introduced a new REST endpoint to retrieve axis information based on UUIDs.
  - Added a new GraphQL query to fetch axis data with transformation details.
  - Enhanced the BivariateMatrix query to include various statistical transformation properties.

- **Enhancements**
  - Added support for detailed statistical transformations in several queries and services.
  - Introduced a new `Transformation` data class to encapsulate statistical properties.

- **Configuration**
  - Updated `insightsApi` URL in the configuration file to point to a localhost endpoint for development.

### CODE REVIEW

File: src/main/java/io/kontur/disasterninja/controller/AxisController.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.UUID;
+
+
+@RestController
+@RequestMapping("/axes")
```
Comment (palina-krukovich) [inline src/main/java/io/kontur/disasterninja/controller/AxisController.java:20]: axis?

File: src/main/java/io/kontur/disasterninja/controller/AxisController.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    @Operation(summary = "Get axis information by numerator and denominator UUID", tags = {"Axis"})
+    @ApiResponse(responseCode = "200",
+            content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE,
+                    schema = @Schema(implementation = BivariateLegendAxisDescription.class)))
+    @GetMapping(path = "/{numerator}/{denominator}")
+    public BivariateLegendAxisDescription get(
+            @PathVariable @Parameter(name = "numerator") UUID numerator,
+            @PathVariable @Parameter(name = "denominator") UUID denominator) {
+        List<BivariateLegendAxisDescription> axes = axisService.getDataForAxis(numerator, denominator);
+        return axes.size() > 0 ? axes.get(0) : null;
```
Comment (palina-krukovich) [inline src/main/java/io/kontur/disasterninja/controller/AxisController.java:35]: We usually try to use ResponseEntity with 204 and 200 statuses for empty and success cases. 204 + empty body would be better than null.

File: src/main/java/io/kontur/disasterninja/controller/AxisController.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    @Operation(summary = "Get axis list", tags = {"Axis"})
+    @ApiResponse(responseCode = "200",
+            content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE,
+                    array = @ArraySchema(schema = @Schema(implementation = BivariateLegendAxisDescription.class))))
+    @GetMapping
+    public List<BivariateLegendAxisDescription> getList() {
+        return axisService.getDataForAxis(null, null);
+    }
```
[the remark this answers is not in the document]
Comment (palina-krukovich) [reply src/main/java/io/kontur/disasterninja/controller/AxisController.java:45]: Here you can also get an empty list. Better to use HTTP statuses

File: src/main/java/io/kontur/disasterninja/domain/Transformation.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package io.kontur.disasterninja.domain;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import lombok.Data;
+
+import java.util.List;
+
+@Data
+@JsonInclude
```
[the remark this answers is not in the document]
Comment (sleeping-h) [reply src/main/java/io/kontur/disasterninja/domain/Transformation.java]: indeed, works fine without it!

File: src/main/java/io/kontur/disasterninja/controller/AxisController.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    @GetMapping(path = "/{numerator}/{denominator}/transformations")
+    public ResponseEntity<List<Transformation>> get(
+            @PathVariable @Parameter(name = "numerator") UUID numerator,
+            @PathVariable @Parameter(name = "denominator") UUID denominator) {
+        List<Transformation> transformations = axisService.getTransformations(numerator, denominator);
+        if (transformations.size() == 0) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(emptyList());
+        }
+        return ResponseEntity.ok(transformations);
+    }
```
[the remark this answers is not in the document]
Comment (sleeping-h) [reply src/main/java/io/kontur/disasterninja/controller/AxisController.java:45]: please provide unit tests

[the remark this answers is not in the document]
Comment (sleeping-h) [reply src/main/java/io/kontur/disasterninja/controller/AxisController.java:45]: i got  error: constructor Transformation in class Transformation cannot be applied to given types;
        List<Transformation> transformations = List.of(new Transformation("scale", 0.0, 1.0, 0.0, 1.0, 0.0, 1.0, List.of(0.0, 1.0)));
                                                       ^
  required: no arguments
  found:    String,double,double,double,double,double,double,List<Double>
  reason: actual and formal argument lists differ in length

you probably should implement a simple constructor for Transformation class

**Review by palina-krukovich — APPROVED** [review]

**Pull request merged**